docs/system-administration/DNS/debian-stretch.md

```diff
@@ -0,0 +1,723 @@
+```
+; <<>> DiG 9.10.3-P4-Raspbian <<>> www.google.com @localhost
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 18246
+;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 13, ADDITIONAL: 27
+
+;; OPT PSEUDOSECTION:
+; EDNS: version: 0, flags:; udp: 4096
+;; QUESTION SECTION:
+;www.google.com.			IN	A
+
+;; ANSWER SECTION:
+www.google.com.		86	IN	A	172.217.24.132
+
+;; AUTHORITY SECTION:
+.			411182	IN	NS	b.root-servers.net.
+.			411182	IN	NS	h.root-servers.net.
+.			411182	IN	NS	g.root-servers.net.
+.			411182	IN	NS	l.root-servers.net.
+.			411182	IN	NS	j.root-servers.net.
+.			411182	IN	NS	f.root-servers.net.
+.			411182	IN	NS	e.root-servers.net.
+.			411182	IN	NS	k.root-servers.net.
+.			411182	IN	NS	a.root-servers.net.
+.			411182	IN	NS	i.root-servers.net.
+.			411182	IN	NS	d.root-servers.net.
+.			411182	IN	NS	m.root-servers.net.
+.			411182	IN	NS	c.root-servers.net.
+
+;; ADDITIONAL SECTION:
+a.root-servers.net.	335582	IN	A	198.41.0.4
+a.root-servers.net.	335424	IN	AAAA	2001:503:ba3e::2:30
+b.root-servers.net.	336203	IN	A	199.9.14.201
+b.root-servers.net.	335567	IN	AAAA	2001:500:200::b
+c.root-servers.net.	335601	IN	A	192.33.4.12
+c.root-servers.net.	335299	IN	AAAA	2001:500:2::c
+d.root-servers.net.	343081	IN	A	199.7.91.13
+d.root-servers.net.	343973	IN	AAAA	2001:500:2d::d
+e.root-servers.net.	336987	IN	A	192.203.230.10
+e.root-servers.net.	336987	IN	AAAA	2001:500:a8::e
+f.root-servers.net.	335460	IN	A	192.5.5.241
+f.root-servers.net.	335895	IN	AAAA	2001:500:2f::f
+g.root-servers.net.	336203	IN	A	192.112.36.4
+g.root-servers.net.	335863	IN	AAAA	2001:500:12::d0d
+h.root-servers.net.	335567	IN	A	198.97.190.53
+h.root-servers.net.	335785	IN	AAAA	2001:500:1::53
+i.root-servers.net.	335363	IN	A	192.36.148.17
+i.root-servers.net.	335424	IN	AAAA	2001:7fe::53
+j.root-servers.net.	335298	IN	A	192.58.128.30
+j.root-servers.net.	335785	IN	AAAA	2001:503:c27::2:30
+k.root-servers.net.	335460	IN	A	193.0.14.129
+k.root-servers.net.	335567	IN	AAAA	2001:7fd::1
+l.root-servers.net.	335601	IN	A	199.7.83.42
+l.root-servers.net.	335443	IN	AAAA	2001:500:9f::42
+m.root-servers.net.	343973	IN	A	202.12.27.33
+m.root-servers.net.	343835	IN	AAAA	2001:dc3::35
+
+;; Query time: 33 msec
+;; SERVER: ::1#53(::1)
+;; WHEN: Sat Nov 21 23:37:26 JST 2020
+;; MSG SIZE  rcvd: 842
+```
+
+Now we have a working caching name server. Time to make the server use it for its own DNS lookups.
+
+### Modifying /etc/resolv.conf
+There seems to be a few ways to do this.
+
+1. Edit the file manually. An acceptable option if you are setting your network statically.
+
+2. Change the name server information at the source. The DHCP server that provides it.
+
+3. Modify **/etc/dhcpcd.conf** if you are assigning host addresses and name servers using dhcp.
+
+4. Using **netplan**. Unfortunately this is new to me and I have not had a chance to look into it.
```

Comment:
Unless you're going to talk about it, maybe worth leaving this option out? The others are very simple methods which I suspect most people will be doing.

Reply:
Sounds fair. I will be looking into in the future for my own understanding. Something that then could be added later.

Reply:
Netplan has been removed.